contracts/cw-proposal-single/src/contract.rs

```diff
@@ -316,8 +316,8 @@ pub fn execute_vote(
     let mut prop = PROPOSALS
         .may_load(deps.storage, proposal_id)?
         .ok_or(ContractError::NoSuchProposal { id: proposal_id })?;
-    if prop.current_status(&env.block) != Status::Open {
-        return Err(ContractError::NotOpen { id: proposal_id });
+    if prop.expiration.is_expired(&env.block) {
+        return Err(ContractError::Expired { id: proposal_id });
```

Comment:
could we add a comment here explaining what is going on?

> Allow voting on proposals until they expire. Voting on a non-open proposal will never change their outcome as if an outcome has been determined, it is because no possible sequence of votes may cause a different one. This then serves to allow for better tallys of opinions in the event that a proposal passes or is rejected early.